components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -120,19 +120,23 @@ public void addCORSOrigins(String applicationId, List<String> origins, String te
         List<Origin> originList = CORSConfigurationUtils.createOriginList(origins);
 
         // Check if the CORS origins are already present.
-        List<CORSOrigin> existingCORSOrigins = getCORSOriginDAO().getCORSOriginsByApplicationId(
-                applicationBasicInfo.getApplicationId(), tenantId);
-        for (Origin origin : originList) {
-            if (existingCORSOrigins.stream().map(CORSOrigin::getId).collect(Collectors.toList())
-                    .contains(origin.getValue())) {
-                // CORS origin is already registered for the application.
-                if (log.isDebugEnabled()) {
-                    log.debug(String.format(ERROR_CODE_ORIGIN_PRESENT.getMessage(), tenantDomain, origin));
+        if (applicationBasicInfo != null) {
+            List<CORSOrigin> existingCORSOrigins = getCORSOriginDAO().getCORSOriginsByApplicationId(
+                    applicationBasicInfo.getApplicationId(), tenantId);
+
+            for (Origin origin : originList) {
+                if (existingCORSOrigins.stream().map(CORSOrigin::getId).collect(Collectors.toList())
```

Comment:
In here, for each `origin` we are creating list of `CORsOrigin` Ids. Can we create one list and check in that list?

Reply:
Fixed